### PR TITLE
Add 2013 MSAMD mappings

### PR DESCRIPTION
**Pull request by m3brown**
2013 records have been added to the msasm.csv file, and the file was re-sorted using the following command:

```shell
sort -n --field-separator=',' -k2 -k1 msamd.csv
```

Also included is a script to ingest data from a CSV or XLSX in the correct format for the msamd.csv file.  For example, to add 2014 records:

```shell
# output the content to stdout for validation
./generate_msamd_csv.sh -y 2014 input-file.csv

# backup the original end file and apply new records
./generate_msamd_csh.sh -y 2014 -a code_sheets/msamd.csv input-file.csv
```

TODO: The script does not handle a couple edge cases with upper/lowercasing:
- Some non-standard letters may not be the correct case
- Some 2 letter words may not have the correct case
  - e.g. with this script `Fond du Lac` would be `Fond Du Lac`

### CODE REVIEW

**Comment by kave** [comment]
@m3brown in your comment "backup the original end file and apply new records" where does it back up the file too? 
Does this command just modify the existing msamd.csv file with the new year?

File: resources/datasets/hmda/generate_msamd_csv.sh [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+                *)
+                    echo "Invalid file type specified" >&2
+                    echo "$OPTIONS_GUIDE" >&2
+                    echo 1
+                    ;;
+            esac
+            ;;
+        y)
+            PREPEND_YEAR=$OPTARG
+            ;;
```
Comment (kave) [inline resources/datasets/hmda/generate_msamd_csv.sh]: what does this do by default if they don't supply a year?

File: resources/datasets/hmda/generate_msamd_csv.sh [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+        echo >&2 "'xlsx2csv' is required for this script to execute. To install, run:
+pip install xlsx2csv 
+Note that root access may be required to run pip commands. Aborting.";
+        exit 1;
+    }
+fi
+
+# NOTE - if using OSX, you'll need to install a newer version of sed to support these queries.
+# `brew install gnu-sed` will create the gsed executable
+sed_bin='sed'
```
Comment (kave) [inline resources/datasets/hmda/generate_msamd_csv.sh]: hmm do we want to do a check for macs here and then print a warning?
e.g. uname -a and check if its a darwin kernel for osx ?

Comment (m3brown) [inline resources/datasets/hmda/generate_msamd_csv.sh]: Good idea.  I couldn't find a good way to check sed version (mac sed doesn't have a '-v' option...) but maybe just saying "if you're a mac and you don't have gsed, warning!"

**Comment by CFPBadmin** [comment]
It backs up the file to filename.timestamp.  It deletes and records that exist for the year you specified, appends the new records to the bottom, then sorts the file to integrate the new year into the original formatting.  I can improved the documentation for this.

File: resources/datasets/hmda/generate_msamd_csv.sh [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+  # remove any line that does not contain a 5 character integer (MSA code)
+  # i.e. remove any header rows
+  /[0-9]\{5\}[,|]/!d;
+
+  # remove the delimiter (| or ,) at the end of every line
+  s/[,|]$//;
+
+  # replace comma separating city block and state block
+  s/, / - /;
+
```
Comment (kave) [inline resources/datasets/hmda/generate_msamd_csv.sh]: couldn't 
```
# remove the delimiter (| or ,) at the end of every line
  s/[,|]$//;

  # replace comma separating city block and state block
  s/, / - /;
```
be merged into 1 line?

Comment (m3brown) [inline resources/datasets/hmda/generate_msamd_csv.sh]: If you have a query to do so, in a way that isn't confusing I'm all for it.  It looks to me like they're two different actions on two different comma scenarios.

Comment (kave) [inline resources/datasets/hmda/generate_msamd_csv.sh]: ooh @m3brown i misread the purpose of the 2nd line. its replacing the commas & dashes with semi-colons. alright maybe we should add that to the comment as well

**Comment by kave** [comment]
@m3brown you've resolved all my questions, approved!

**Pull request merged**